spec/labels_checker_spec.rb

```diff
@@ -0,0 +1,173 @@
+# frozen_string_literal: true
+
+require_relative 'spec_helper'
+
+module Danger
+  describe Danger::LabelsChecker do
+    it 'is a plugin' do
+      expect(described_class.new(nil)).to be_a Danger::Plugin
+    end
+
+    describe 'with Dangerfile' do
+      before do
+        @dangerfile = testing_dangerfile
+        @plugin = @dangerfile.labels_checker
+      end
+
+      context 'with required labels' do
+        it 'returns a custom error when a PR does not have at least one label' do
+          allow(@plugin.github).to receive(:pr_labels).and_return([])
+
+          error = 'PR is missing at least one label.'
+          @plugin.check(required_labels: [/.*/], required_labels_error: error)
```

Comment:
Note that the RegExp are not anchored in Ruby, so they can match just part of the string instead of the whole string. Which that means that `/x/.match?('abcxdef')` will match too.

As a result:
 - We should use `^` and `$` to anchor the regex if we want to make sure that `/feature: .*/` won't risk matching a label like `'secret feature: admin mode'`.
 - Or if the goal is just to match labels that starts with a specific string, we don't need to use `.*$` to match up to the end, and we can just use `/^feature:/` for the RegExp
 - And similarly here, using `//` as an empty RegExp should match _any_ string (i.e. `//.match?(text)` will always return true), so you could use that instead of `/.*/`.

Reply:
Nice observation 👍